modules/guides/pages/import.adoc

```diff
@@ -0,0 +1,335 @@
+= Import Data
+// Import and Stream Data
+:page-topic-type: guide
+:imagesdir: ../assets/images
+:tabs:
+:description: How to import documents into Couchbase server.
+
+[abstract]
+{description}
+
+== Introduction
+
+Importing data can be done from the Couchbase Server UI,
+via the `cbimport` command-line tool shipped with Couchbase Server,
+or using the SDK to script the process.
+
+XXX server/current/tools/cbimport.html
+
+Data load essentially consists of:
+
+* Prepare data in some well known format such as Comma Separated Values or a stream of JSON documents
+
+* Connect to the Couchbase instance
+
+* Iterate over every document in the data load
+
+  * Decide on the key for this document (could be an ID, a sequence number, or some combination of fields)
+  
+  * Do any additional processing
+
+  * Upsert the document
+
+
+== Preparing the Data
+
+Extract or generate your data in the appropriate data format. The following are well supported for export as well as by `cbimport` and the module ecosystems of all our SDKs.
+
+
+[{tabs}]
+====
+CSV::
++
+--
+Comma Separated Values (.csv) are easily exported from many spreadsheet and database applications.
+
+Typically, the first row is a header row containing the names of the columns within the document.
```

Comment:
Ah. CSV lets you do either, but the *code examples* use the header row, because I've made decision that's the best approach for us.

Just checking we can be *opinionated* here? And I don't have to pepper with `NOTE:`s pointing out that you could do it differently...

Reply:
Hmmm, it's an arguable question at this specific point in the document — are you describing the layout of a typical CSV file here, or instructing users how to lay out a CSV file for import?

Generally the example files should illustrate the instructions, rather than the other way around.

It's fine to be opinionated — in fact you have to be, because we have to decide what procedures we're going to cover, and what steps each procedure should include. The instructions should be clear and concise, so we can only cover the most likely use cases, referring to the reference material for everything else.

The point to bear in mind is that in a how-to guide, we don't know exactly what the user is trying to do. They have a problem they're trying to solve: import some data. So (IMHO) we can't say to them, "our file looks like this, we want to do this". We don't know what their file looks like. We have to give them instructions that will cover as much of their problem as is reasonably possible. The examples to illustrate the procedure and confirm that they're on the right lines. Then refer them to the reference pages for edge cases.